src/Adapters/Auth/auth0.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+// tenantId: the auth0 tenant, e.g. example.eu.auth0.com (NO HTTP / HTTPS)
+// clientId: the auth0 client id, can be found in the auth0 console.
+
+var Parse = require('parse/node').Parse;
+const jwksClient = require('jwks-rsa');
+const jwt = require('jsonwebtoken');
+const util = require('util');
+
+const getAuth0KeyByKeyId = async (keyId, cacheMaxEntries, cacheMaxAge, tenantId) => {
+  const client = jwksClient({
+    jwksUri: `https://${tenantId}/.well-known/jwks.json`,
+    cache: true,
+    cacheMaxEntries,
+    cacheMaxAge,
+  });
+
+  const asyncGetSigningKeyFunction = util.promisify(client.getSigningKey);
+
+  let key;
+  try {
+    key = await asyncGetSigningKeyFunction(keyId);
```

Comment:
Prefer `return await asyncGetSigningKeyFunction(keyId);`, no need to define `key`